scipy/optimize/optimize.py

```diff
@@ -2397,20 +2399,76 @@ def bracket(func, xa=0.0, xb=1.0, args=(), grow_limit=110.0, maxiter=1000):
     return xa, xb, xc, fa, fb, fc, funcalls
 
 
-def _linesearch_powell(func, p, xi, tol=1e-3):
+def _line_for_search(x0, alpha, lower_bound, upper_bound):
```

Comment:
`_line_for_search` needs to have a test suite in `test_optimize`

Reply:
Done. `_line_for_search` is also tested via the tests for `_linesearch_powell`.